src/IR/DropInference.hs

```diff
@@ -100,12 +100,15 @@ insertDropExpr :: I.Expr Poly.Type -> InsertFn (I.Expr Poly.Type)
 
 -- | Inserting dup/drops into function application with arg
 insertDropExpr (I.App fun arg typ) = do
-  -- fun' <- insertDropExpr fun
   arg' <- insertDropExpr arg
   return $ I.App fun arg' typ
 
+-- | Skip let bindings that have unit type
```

Comment:
You should remove the `|` from the comment here. That is used to start haddock documentation for top-level definitions, and should only be used one per definition.